polaris-core/src/main/java/org/apache/polaris/core/persistence/PrincipalSecretsGenerator.java

```diff
@@ -82,4 +86,28 @@ static PrincipalSecretsGenerator bootstrap(String realmName, Function<String, St
       }
     };
   }
+
+  /** Return true if environment variables for client ID & secret are set */
+  static boolean hasCredentialVariables(String realmName, String principalName) {
```

Comment:
Exposing this `static` method does not look nice to be from the encapsulation point of view. I'd prefer to keep credential generation logic inside `PrincipalSecretsGenerator` instances.

We already have the `bootstrap` flag in `LocalPolarisMetaStoreManagerFactory` that controls secret generation under a synchronization lock. WDYT about converting it into something like `PrincipalSecretsGenerator generatorFallback` and making `PrincipalSecretsGenerator.bootstrap()` take an explicit fallback parameter (which is currently always `RANDOM_SECRETS`)? `LocalPolarisMetaStoreManagerFactory` would use `RANDOM_SECRETS` as the fallback by default, but during bootstrapping, we could have a custom impl. that produces random secrets if printing is enabled, but throws an exception if fallback if requested, but printing is disabled.